src/esw/brushed_motors/Controller.cpp

```diff
@@ -197,12 +198,45 @@ void Controller::turnOn() const {
 }
 
 // REQUIRES: nothing
-// MODIFIES: isLive
+// MODIFIES: nothing
+// EFFECTS: Returns a combined ID for both the deviceAddress and motorID
+uint8_t Controller::combineDeviceMotorID() const {
+    return (deviceAddress << 3) | motorID;
+}
+
+// REQUIRES: nothing
+// MODIFIES: liveMap
 // EFFECTS: I2C bus, if not already live,
 // configures the physical controller.
 // Then makes live.
+
+std::unordered_map<uint8_t, LiveState> Controller::liveMap;
+
 void Controller::makeLive() {
-    if (isLive) {
+    // if (isLive) {
+    //     return;
+    // }
+
+    uint8_t key = combineDeviceMotorID();
+
+    // if key is absent, no motor with that key has been made live
+    auto it = liveMap.find(key);
+    if (it != liveMap.end()) {
+        std::unique_lock<std::mutex>
+            lck(it->second.liveMutex);
+        // it->second.liveMutex.lock();
+    } else {
+        LiveState state;
+        state.jointName = name;
+        liveMap.emplace(key, state);
+        // liveMap[key] = state; // doesn't work anymore because mutex isn't inline static
+        std::unique_lock<std::mutex>
+            lck(state.liveMutex);
+        // state.liveMutex.lock();
+    }
+
+    // already live and configured to correct motor
```

Comment:
Oh, based on the logic from previously, this would always be true on startup (right after you emplace a key and state) so you might want another condition for that case.